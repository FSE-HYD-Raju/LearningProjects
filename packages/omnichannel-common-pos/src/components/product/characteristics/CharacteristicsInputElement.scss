.CharacteristicsInputElement {
	&-container {
		width: 100%;
	}
}
