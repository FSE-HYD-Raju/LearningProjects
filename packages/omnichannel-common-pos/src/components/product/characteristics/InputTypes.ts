enum InputTypesEnum {
	DATE = "date",
	TIME = "time",
	DATE_TIME = "date-time",
	DATE_TIME_PERIOD = "date-time-period",
	TIME_OF_THE_DAY_PERIOD = "time-of-the-day-period",
	DROPDOWN = "dropdown",
	RADIO = "radio",
	TEXT = "text",
}

type InputTypes = keyof typeof InputTypesEnum;
export {
	InputTypesEnum,
	InputTypes,
};
