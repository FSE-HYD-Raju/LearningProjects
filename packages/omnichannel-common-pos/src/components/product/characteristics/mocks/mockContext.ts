import TestUtils from "../../../../testUtils";

const mockContext = {
	flux: {
		actions: {
			BasketActions: {
				resetValidIcc: jest.fn(),
				triggerICCIDPreactivationValidation: jest.fn(),
				resetValidFnF: jest.fn(),
				triggerFnfValidation: jest.fn()
			}
		},
	},
	store: TestUtils.mockReduxStore({
		feature: {
			characteristicsAliases: {},
			ICCIDPreactivationValidationPOs: [],
		}
	})
};

export default mockContext;
