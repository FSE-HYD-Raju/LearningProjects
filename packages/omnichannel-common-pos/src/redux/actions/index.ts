import { Action } from "redux";
import * as types from "../types";

import { auth, AuthActions } from "../models/auth/auth.actions";
export { auth, AuthActions } from "../models/auth/auth.actions";

import { b2cCheckout, B2CCheckoutActions } from "../models/b2cCheckout/b2cCheckout.actions";
export { b2cCheckout, B2CCheckoutActions } from "../models/b2cCheckout/b2cCheckout.actions";

import { base, BaseActions } from "../models/base/base.actions";
export { base, BaseActions } from "../models/base/base.actions";

import { basket, BasketActions } from "../models/basket/basket.actions";
export { basket, BasketActions } from "../models/basket/basket.actions";

import { category, CategoryActions } from "../models/category/category.actions";
export { category, CategoryActions } from "../models/category/category.actions";

import { cmsAdmin, CmsAdminActions } from "../models/cmsAdmin/cmsAdmin.actions";
export { cmsAdmin, CmsAdminActions } from "../models/cmsAdmin/cmsAdmin.actions";

import { cms, CmsActions } from "../models/cms/cms.actions";
export { cms, CmsActions } from "../models/cms/cms.actions";

import { comparison, ComparisonActions } from "../models/comparison/comparison.actions";
export { comparison, ComparisonActions } from "../models/comparison/comparison.actions";

import { consul, ConsulActions } from "../models/consul/consul.actions";
export { consul, ConsulActions } from "../models/consul/consul.actions";

import { currency, CurrencyActions } from "../models/currency/currency.actions";
export { currency, CurrencyActions } from "../models/currency/currency.actions";

import { customer, CustomerActions } from "../models/customer/customer.actions";
export { customer, CustomerActions } from "../models/customer/customer.actions";

import { customerCase, CustomerCaseActions } from "../models/customerCase/customerCase.actions";
export { customerCase, CustomerCaseActions } from "../models/customerCase/customerCase.actions";

import { delivery, DeliveryActions } from "../models/delivery/delivery.actions";
export { delivery, DeliveryActions, DeliveryActionPayload } from "../models/delivery/delivery.actions";

import { changeSim, ChangeSimActions } from "../models/eCare/changeSim/changeSim.actions";
export { changeSim, ChangeSimActions, ChangeSimActionPayload } from "../models/eCare/changeSim/changeSim.actions";

import { changePlan, ChangePlanActions } from "../models/eCare/changeTariffPlan/changePlan.actions";
export { changePlan, ChangePlanActions, ChangePlanActionPayload } from "../models/eCare/changeTariffPlan/changePlan.actions";

import { recurringTopUp, RecurringTopUpActions } from "../models/eCare/recurringTopUp/recurringTopUp.actions";
export { recurringTopUp, RecurringTopUpActions, RecurringTopUpActionPayload } from "../models/eCare/recurringTopUp/recurringTopUp.actions";

import { digitalLife, DigitalLifeActions } from "../models/digitalLife/digitalLife.actions";
export { digitalLife, DigitalLifeActions, DigitalLifeActionPayload } from "../models/digitalLife/digitalLife.actions";

import { activateSim, ActivateSimActions } from "../models/eCare/activateSim/activateSim.actions";
export { activateSim, ActivateSimActions, ActivateSimActionPayload } from "../models/eCare/activateSim/activateSim.actions";

import { suspension, SuspensionActions } from "../models/eCare/suspension/suspension.actions";
export { suspension, SuspensionActions, SuspensionActionPayload } from "../models/eCare/suspension/suspension.actions";

import { document, DocumentActions } from "../models/document/document.actions";
export { document, DocumentActions } from "../models/document/document.actions";

import { error, ErrorActions } from "../models/error/error.actions";
export { error, ErrorActions } from "../models/error/error.actions";

import { basketError, BasketErrorActions } from "../models/basketError/basketError.actions";
export { basketError, BasketErrorActions } from "../models/basketError/basketError.actions";

import { feature, FeatureActions } from "../models/feature/feature.actions";
export { feature, FeatureActions } from "../models/feature/feature.actions";

import { lifecycle, LifecycleActions } from "../models/lifecycle/lifecycle.actions";
export { lifecycle, LifecycleActions } from "../models/lifecycle/lifecycle.actions";

import { loadingOverlay, LoadingOverlayActions } from "../models/loadingOverlay/loadingOverlay.actions";
export { loadingOverlay, LoadingOverlayActions } from "../models/loadingOverlay/loadingOverlay.actions";

import { location, LocationActions } from "../models/location/location.actions";
import { LocationItem } from "../models/location/location.types";
export { location, LocationActions } from "../models/location/location.actions";

import { msisdn, MsisdnActions } from "../models/msisdn/msisdn.actions";
export { msisdn, MsisdnActions } from "../models/msisdn/msisdn.actions";

import { navBar, NavBarActions } from "../models/navBar/navBar.actions";
export { navBar, NavBarActions } from "../models/navBar/navBar.actions";

import { notification, NotificationActions } from "../models/notification/notification.actions";
export { notification, NotificationActions } from "../models/notification/notification.actions";

import { Identification } from "../types/Identification";
import { organization, OrganizationActions } from "../models/organization/organization.actions";
export { organization, OrganizationActions } from "../models/organization/organization.actions";

import { payment, PaymentActions } from "../models/payment/payment.actions";
export { payment, PaymentActions } from "../models/payment/payment.actions";

import { portIn, PortInActions } from "../models/portIn/portIn.actions";
export { portIn, PortInActions } from "../models/portIn/portIn.actions";

import { marketingConsent, MarketingConsentActions } from "../models/marketingConsent/marketingConsent.actions";
export { marketingConsent, MarketingConsentActions } from "../models/marketingConsent/marketingConsent.actions";

import { posCheckout, PosCheckoutActions } from "../models/posCheckout/posCheckout.actions";
export { posCheckout, PosCheckoutActions } from "../models/posCheckout/posCheckout.actions";

import {
	productOfferingConfiguration,
	ProductOfferingConfigurationActions
} from "../models/productOfferingConfiguration/productOfferingConfiguration.actions";
export {
	productOfferingConfiguration,
	ProductOfferingConfigurationActions
} from "../models/productOfferingConfiguration/productOfferingConfiguration.actions";

import { productLoan, ProductLoanActions } from "../models/productLoan/productLoan.actions";
export { productLoan, ProductLoanActions } from "../models/productLoan/productLoan.actions";

import { sales, SalesActions } from "../models/sales/sales.actions";
export { sales, SalesActions } from "../models/sales/sales.actions";

import { salesRepSession, SalesRepSessionActions } from "../models/salesRepSession/salesRepSession.actions";
export { salesRepSession, SalesRepSessionActions } from "../models/salesRepSession/salesRepSession.actions";

import { schema, SchemaActions } from "../models/schema/schema.actions";
export { schema, SchemaActions } from "../models/schema/schema.actions";

import { service, ServiceActions } from "../models/service/service.actions";
export { service, ServiceActions } from "../models/service/service.actions";

import { session, SessionActions } from "../models/session/session.actions";
export { session, SessionActions } from "../models/session/session.actions";

import { toaster, ToasterActions } from "../models/toaster/toaster.actions";
export { toaster, ToasterActions } from "../models/toaster/toaster.actions";

import { translation, TranslationActions } from "../models/translation/translation.actions";
export { translation, TranslationActions } from "../models/translation/translation.actions";

import { uriLocation, UriLocationActions } from "../models/uriLocation/uriLocation.actions";
export { uriLocation, UriLocationActions } from "../models/uriLocation/uriLocation.actions";

import { user, UserActions } from "../models/user/user.actions";
export { user, UserActions } from "../models/user/user.actions";

import { versionInformation, VersionInformationActions } from "../models/versionInformation/versionInformation.actions";
export { versionInformation, VersionInformationActions } from "../models/versionInformation/versionInformation.actions";

import { eligibility, EligibilityActions } from "../models/eligibility/eligibility.actions";
import { EligibilityDecisionUseCase, RecipeId } from "../models/eligibility/eligibility.types";
export { eligibility, EligibilityActions } from "../models/eligibility/eligibility.actions";

import { productOfferings, ProductOfferingsActions } from "../models/productOfferings/productOfferings.actions";
import {
	QueryStates,
	ProductOfferings,
	ProductOfferingsState
} from "../models/productOfferings/productOfferings.types";
import { Cardinality, CommercialEnrichments, ProductOfferingAttributes } from "../types";
export { productOfferings, ProductOfferingsActions } from "../models/productOfferings/productOfferings.actions";

import { LocationDescriptorObject } from "history";
export { LocationDescriptorObject } from "history";

import { routerActions } from "connected-react-router";

import {
	customerInteractions,
	CustomerInteractionsActions
} from "../models/customerInteractions/customerInteractions.actions";
export {
	customerInteractions,
	CustomerInteractionsActions,
	CustomerInteractionsActionPayload
} from "../models/customerInteractions/customerInteractions.actions";

import { support, SupportActions } from "../models/support/support.actions";
import { ChangeSimServiceSubmitData } from "../services/ChangeSimService";
import { ErrorForModal, Error } from "../services/ErrorContainer";
export { support, SupportActions } from "../models/support/support.actions";

import { MsisdnSelectionUseCase } from "../models/msisdnSelection/msisdnSelection.types";
import { msisdnSelection, MsisdnSelectionActions } from "../models/msisdnSelection/msisdnSelection.actions";
import { InitializeAddonConfig } from "../services/AddonService";
export { msisdnSelection, MsisdnSelectionActions } from "../models/msisdnSelection/msisdnSelection.actions";
import { ProvinceAndCityPayload, provinceAndCity, ProvinceAndCityActions } from "../models/provinceAndCity/provinceAndCity.actions";
export { ProvinceAndCityPayload, provinceAndCity, ProvinceAndCityActions } from "../models/provinceAndCity/provinceAndCity.actions";
import { workforce, WorkforceActions } from "../models/workforce/workforce.actions";
export { workforce, WorkforceActions } from "../models/workforce/workforce.actions";

const actions = {
	auth,
	b2cCheckout,
	base,
	basket,
	category,
	cmsAdmin,
	cms,
	comparison,
	consul,
	currency,
	customerCase,
	customer,
	delivery,
	changeSim,
	changePlan,
	recurringTopUp,
	digitalLife,
	activateSim,
	suspension,
	document,
	eligibility,
	error,
	basketError,
	feature,
	lifecycle,
	loadingOverlay,
	location,
	marketingConsent,
	msisdn,
	navBar,
	notification,
	organization,
	posCheckout,
	payment,
	portIn,
	productOfferingConfiguration,
	productOfferings,
	productLoan,
	provinceAndCity,
	router: routerActions,
	sales,
	salesRepSession,
	schema,
	service,
	session,
	support,
	toaster,
	translation,
	uriLocation,
	user,
	versionInformation,
	customerInteractions,
	msisdnSelection,
	workforce,
};

export type AppActions = typeof actions;

export default actions;
