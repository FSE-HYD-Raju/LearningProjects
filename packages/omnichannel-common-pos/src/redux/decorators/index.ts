"use strict";

export { deprecated } from "./deprecated";
