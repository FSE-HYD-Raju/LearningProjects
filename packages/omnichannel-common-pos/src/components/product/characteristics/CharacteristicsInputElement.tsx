import cssns from "../../../utils/cssnsConfig";
import CharacteristicsTextInputContainer from "./CharacteristicsTextInputContainer";
import CharacteristicsDropdown from "./CharacteristicsDropdown";
import CharacteristicsRadioInput from "./CharacteristicsRadioInput";
import CharacteristicsDate from "./CharacteristicsDate";
import { InputTypesEnum } from "./InputTypes";
import { FC } from "react";
import { POSFunctionCustomizationPoints, withCustomization } from "../../../customization";
import CharacteristicsNotifications from "./CharacteristicsNotifications";
import {
	Characteristic,
	ProductOffering,
	ProductPath,
	NominationCharacteristics
} from "../../../redux/types";
import { ContextType, contextTypesValidationMap } from "../../../types";

const { React } = cssns("CharacteristicsInputElement");

interface CharacteristicsInputElementProps {
	productOfferingErrors?: Array<any>; // TODO: add types here
	clearProductOfferingErrors: () => void;
	path: ProductPath;
	inputCharacteristic?: Characteristic;
	characteristicKey: string;
	inputtedCharacteristics?: Record<string, string>;
	enhancedCharacteristics?: Record<string, Array<any>>;
	isMandatory?: boolean;
	validation?: string;
	icc_display_mode?: string;
	product: ProductOffering;
	key?: string;
	nominationPOCharacteristics: NominationCharacteristics | null;
	isNominationPO?: boolean;
}

const hasValues = (values?: Array<any>) => {
	return Array.isArray(values) && values.length > 0;
};

// TODO: check if this is valid (in runtime)
const isDateType = (inputCharacteristic?: Characteristic): boolean => {
	const dataType = inputCharacteristic && inputCharacteristic.dataType;
	return (
		dataType === InputTypesEnum.DATE_TIME_PERIOD ||
		dataType === InputTypesEnum.DATE ||
		dataType === InputTypesEnum.DATE_TIME ||
		dataType === InputTypesEnum.TIME ||
		dataType === InputTypesEnum.TIME_OF_THE_DAY_PERIOD
	);
};

const getDateType = (inputCharacteristic: Characteristic): string | undefined => {
	const dataType = inputCharacteristic.dataType;
	if (dataType === InputTypesEnum.DATE) {
		return InputTypesEnum.DATE;
	} else if (dataType === InputTypesEnum.DATE_TIME_PERIOD) {
		return InputTypesEnum.DATE_TIME_PERIOD;
	} else if (dataType === InputTypesEnum.DATE_TIME) {
		return InputTypesEnum.DATE_TIME;
	} else if (dataType === InputTypesEnum.TIME) {
		return InputTypesEnum.TIME;
	} else if (dataType === InputTypesEnum.TIME_OF_THE_DAY_PERIOD) {
		return InputTypesEnum.TIME_OF_THE_DAY_PERIOD;
	}
	return undefined;
};

const getInputType = (props: CharacteristicsInputElementProps) => {
	const { inputCharacteristic, icc_display_mode } = props;
	if (inputCharacteristic && inputCharacteristic.inputType === "dropdown") {
		return InputTypesEnum.DROPDOWN;
	}
	if (isDateType(inputCharacteristic)) {
		return getDateType(inputCharacteristic!);
	}
	const values = inputCharacteristic && inputCharacteristic.values;
	if (hasValues(values) && icc_display_mode === InputTypesEnum.DROPDOWN) {
		return InputTypesEnum.DROPDOWN;
	}
	if (hasValues(values) && icc_display_mode === InputTypesEnum.RADIO) {
		return InputTypesEnum.RADIO;
	} else {
		return InputTypesEnum.TEXT;
	}
};

const BaselineRenderCharacteristicsInputElement = (props: CharacteristicsInputElementProps, context: ContextType) => {
	const { inputCharacteristic } = props;
	const inputType = getInputType(props);

	if (inputCharacteristic && inputCharacteristic.hidden) {
		return null;
	}

	if (isDateType(inputCharacteristic)) {
		return (
			 <CharacteristicsDate
				flux={context.flux}
				path={props.path}
				characteristicKey={props.characteristicKey}
				inputCharacteristic={props.inputCharacteristic}
				inputType={inputType}
			/>
		);
	}

	switch (inputType) {
		case InputTypesEnum.DROPDOWN:
			return (
				<CharacteristicsDropdown
					flux={context.flux}
					path={props.path}
					inputCharacteristic={props.inputCharacteristic}
					characteristicKey={props.characteristicKey}
					enhancedCharacteristics={props.enhancedCharacteristics}
					inputtedCharacteristics={props.inputtedCharacteristics}
					product={props.product}
				/>
			);
		case InputTypesEnum.RADIO:
			return (
				<CharacteristicsRadioInput
					flux={context.flux}
					characteristicKey={props.characteristicKey}
					inputtedCharacteristics={props.inputtedCharacteristics}
					product={props.product}
					path={props.path}
					inputCharacteristic={props.inputCharacteristic}
				/>
			);
		default:
			return (
				<CharacteristicsTextInputContainer
					flux={context.flux}
					path={props.path}
					isMandatory={props.isMandatory}
					inputCharacteristic={props.inputCharacteristic}
					characteristicKey={props.characteristicKey}
					inputtedCharacteristics={props.inputtedCharacteristics}
					product={props.product}
					nominationPOCharacteristics={props.nominationPOCharacteristics}
					isNominationPO={props.isNominationPO}
				/>
			);
	}
};

BaselineRenderCharacteristicsInputElement.contextTypes = contextTypesValidationMap;

const RenderCharacteristicsInputElementFunction = withCustomization<CharacteristicsInputElementProps>(
	POSFunctionCustomizationPoints.RENDER_CHARACTERISTICS_INPUT_ELEMENT,
	BaselineRenderCharacteristicsInputElement
);

const CharacteristicsInputElement: FC<CharacteristicsInputElementProps> = (props: CharacteristicsInputElementProps) => {
	return (
		<div className="container">
			<RenderCharacteristicsInputElementFunction {...props}/>
			<CharacteristicsNotifications
				mappedProductOfferingErrors={props.productOfferingErrors}
				targetCharacteristic={props.characteristicKey}
				clearProductOfferingErrors={props.clearProductOfferingErrors}
			/>
		</div>
	);
};

export default CharacteristicsInputElement;
export {
	CharacteristicsInputElementProps,
};
