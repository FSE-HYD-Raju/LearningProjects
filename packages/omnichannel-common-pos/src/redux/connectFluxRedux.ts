"use strict";

import { isFunction, isEqual, pick, chain, values } from "lodash";
import { Store, Unsubscribe } from "redux";
import { AppActions } from "./actions";
import { AppState } from "./reducers";
import {
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration
} from "./types";
import { ConnectFluxReduxType } from "./types/ConnectFluxReduxType";

import { isLogsOn, isDev } from "./loggerMiddleware";

export const getDefaultFluxStoreMap = (): FluxReduxMap<AppState> => {
	return {
		B2CCheckoutStore: "b2cCheckout",
		BaseStore: "base",
		BasketStore: "basket",
		CMSAdminStore: "cmsAdmin",
		ConsulStore: "consul",
		CustomerCaseStore: "customerCase",
		CustomerStore: "customer",
		DigitalLifeStore: "digitalLife",
		CustomerInteractionsStore: "customerInteractions",
		MsisdnStore: "msisdn",
		POSCheckoutStore: "posCheckout",
		PaymentStore: "payment",
		SalesRepSessionStore: "salesRepSession",
		SalesStore: "sales",
		UserStore: "user",
	};
};

const observeFluxStore = <T, A>(fluxStore: FluxStore<T>,
								store: Store<AppState>,
								targetStoreAction: ReduxFluxActionMigration): Unsubscribe => {
	return fluxStore.listen((state: any) => {
		const pureState = filterEnumerable(state);
		store.dispatch(targetStoreAction.fluxSync(pureState));
	});
};

const observeReduxStore = (store: Store<AppState>,
						   select: (state: AppState) => AppState,
						   onChange: (data: Object) => void): Unsubscribe => {

	let currentState: Object = {};

	return store.subscribe(() => {
		const newState = select(store.getState());
		if (!isEqual(newState, currentState)) {
			currentState = {...newState};
			onChange(currentState);
		}
	});
};

const handleReduxFluxChange = <T>(fluxStore: FluxStore<T>, changed: Object, appStateProp: string, key: string) => {
	const oldState = fluxStore.getState();
	const pureState = filterEnumerable(oldState);
	if (!isEqual(pureState, changed)) {
		printLog(`${appStateProp} -> ${key}: `, pureState, "changed", changed);
		setFluxState(fluxStore, changed);
	}
};

export const connectFluxRedux: ConnectFluxReduxType = <T extends AppState, A extends AppActions>(
	flux: FluxAlt<T>,
	store: Store<AppState>,
	fluxMap: Partial<FluxReduxMap<T>>,
	actions: Partial<A>
): Disposers => {

	const state = store.getState();

	const disposers = Object
		.keys(fluxMap)
		.reduce((subs: Disposers, key: string): Disposers|any => {
			const appStateProp = fluxMap[key] as keyof AppState;
			const fluxStore = flux.stores[key];
			const targetStoreAction = actions[appStateProp] as ReduxFluxActionMigration;

			if (isStoresValid(key, fluxStore, targetStoreAction, state, appStateProp)) {
				// Flux -> Redux
				// Connect and reflect changes from flux store in redux state
				subs.fluxDisposers[key] = observeFluxStore(fluxStore, store, targetStoreAction);
				// Redux -> Flux
				// Connect and reflect changes from redux state in flux store
				subs.reduxDisposers[appStateProp] = observeReduxStore(
					store,
					(state: AppState): any => state[appStateProp],
					(changed: Object) => handleReduxFluxChange(fluxStore, changed, appStateProp, key)
				);
			}

			return subs;
		}, {
			fluxDisposers: {},
			reduxDisposers: {}
		});

	return Disposable(disposers);

};

function setFluxState<T>(store: FluxStore<T>, newState: T) {
	store.config.setState.call(store, store.state, newState);
	store.emitChange();
}

function filterEnumerable(obj: Object) {
	return pick(obj, Object.keys(obj));
}

function isStoresValid(key: string,
					   fluxStore: FluxStore<any>,
					   targetStoreAction: any,
					   state: AppState,
					   appStateProp: keyof AppState): boolean {
	const warnings = [];
	if (!fluxStore) {
		warnings.push(`key "${key}" does not exist in flux.stores`);
	}

	if (!targetStoreAction || !isFunction(targetStoreAction.fluxSync)) {
		warnings.push(`fluxSync is not a function in store for ${key}: `, targetStoreAction);
	}

	if (!state[appStateProp]) {
		warnings.push(`app state for "${appStateProp}" does not exist, please check reducers`);
	}

	if (warnings.length) {
		console.info(...warnings);
	}

	return Boolean(fluxStore && targetStoreAction && targetStoreAction.fluxSync && state[appStateProp]);
}

function printLog(action: string, ...data: any[]): void {
	if (!isDev) {
		return;
	}

	if (isLogsOn()) {
		console.groupCollapsed(action);
		console.info(
			"%c state from redux",
			"color: #4CAF50; font-weight: bold",
			...data
		);
		console.groupEnd();
	}
}

function Disposable(disposers: Disposers): Disposers {
	return {
		...disposers,
		disposeAll() {
			chain(disposers)
				.map(values)
				.flatten()
				.forEach(dispose => dispose())
				.value();
		}
	};
}
