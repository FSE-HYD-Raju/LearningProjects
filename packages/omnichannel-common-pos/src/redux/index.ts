"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
