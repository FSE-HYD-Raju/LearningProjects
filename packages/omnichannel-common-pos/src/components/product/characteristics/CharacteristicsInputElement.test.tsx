import * as React from "react";
import {
	shallowWithContext,
	mountWithContext,
} from "../../../testUtils";
import CharacteristicsInputElement from "./CharacteristicsInputElement";
import { Characteristic, NominationCharacteristics, ProductOffering } from "../../../redux";
import mockContext from "./mocks/mockContext";
import { InputTypesEnum } from "./InputTypes";

describe("CharacteristicsInputElement", () => {
	const inputCharacteristic = {
		values: [
			{ name: "Awesome", value: "100" },
			{ name: "Super awesome", value: "1000" },
			{ name: "Not awesome", value: "1" }
		]
	} as any as Characteristic;
	const minProps = {
		clearProductOfferingErrors: jest.fn(),
		setInputtedCharacteristic: jest.fn(),
		inputCharacteristic,
		path: [],
		characteristicKey: "CH",
		inputtedCharacteristics: {},
		icc_display_mode: "aaa",
		product: {} as any as ProductOffering,
		nominationPOCharacteristics: {} as any as NominationCharacteristics,
		isNominationPO: false,
	};

	it("should shallow mount with min props", () => {
		const wrapper = shallowWithContext(<CharacteristicsInputElement {...minProps} />, { context: mockContext });
		expect(wrapper).toMatchSnapshot();
	});

	it("should render dropdown if inputCharacteristic has values", () => {
		const props = {
			...minProps,
			icc_display_mode: InputTypesEnum.DROPDOWN,
		};
		const wrapper = mountWithContext(<CharacteristicsInputElement {...props}/>, { context: mockContext });
		const options = wrapper.find("option");
		expect(options.length).toBe(4);
	});

	it("should render radio if inputCharacteristic has values and sub-type is radio", () => {
		const props = {
			...minProps,
			icc_display_mode: "radio",
		};
		const wrapper = mountWithContext(<CharacteristicsInputElement {...props}/>, { context: mockContext });
		const input = wrapper.find("[type=\"radio\"]");
		expect(input.exists()).toBeTruthy();
	});

	it("should render date time period if inputCharacteristic data type is date-time-period", () => {
		const props = {
			...minProps,
			inputCharacteristic: {
				dataType: InputTypesEnum.DATE_TIME_PERIOD,
			} as any as Characteristic,
		};

		const wrapper = mountWithContext(<CharacteristicsInputElement {...props}/>, { context: mockContext });
		const datePicker = wrapper.find(".OcDatePicker");
		expect(datePicker.exists()).toBeTruthy();
	});
});
