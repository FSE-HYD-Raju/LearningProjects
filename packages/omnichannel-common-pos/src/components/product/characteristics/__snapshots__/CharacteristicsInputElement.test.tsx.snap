// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacteristicsInputElement should shallow mount with min props 1`] = `
<div
  className="CharacteristicsInputElement-container"
>
  <BaselineRenderCharacteristicsInputElement
    characteristicKey="CH"
    clearProductOfferingErrors={[MockFunction]}
    icc_display_mode="aaa"
    inputCharacteristic={
      Object {
        "values": Array [
          Object {
            "name": "Awesome",
            "value": "100",
          },
          Object {
            "name": "Super awesome",
            "value": "1000",
          },
          Object {
            "name": "Not awesome",
            "value": "1",
          },
        ],
      }
    }
    inputtedCharacteristics={Object {}}
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    isNominationPO={false}
    nominationPOCharacteristics={Object {}}
    path={Array []}
    product={Object {}}
    setInputtedCharacteristic={[MockFunction]}
  />
  <CharacteristicsNotifications
    clearProductOfferingErrors={[MockFunction]}
    targetCharacteristic="CH"
  />
</div>
`;
